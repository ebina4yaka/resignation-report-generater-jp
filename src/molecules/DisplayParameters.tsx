import React, { useContext } from 'react'
import { createStyles, makeStyles } from '@material-ui/core/styles'
import { parametersContext } from '../context/useParameters'
import Parameters from '../atoms/Parameter'
import convertDateStringToJapanFormat from '../libs/convertDateStringToJapanFormat'

const useStyles = makeStyles(
  createStyles({
    root: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'column',
      justifyContent: 'center',
    },
  })
)

export default function DisplayParameters(): React.ReactElement {
  const context = useContext(parametersContext)
  const {
    name,
    companyName,
    department,
    representativeDirector,
    dateOfRetirement,
    reason,
    daysOfPaidLeaveRemaining,
  } = context
  const classes = useStyles()
  return (
    <div className={classes.root}>
      <Parameters name="退職者" value={name} />
      <Parameters name="会社名" value={companyName} />
      <Parameters name="所属部署" value={department} />
      <Parameters name="代表取締役" value={representativeDirector} />
      <Parameters
        name="退職日"
        value={convertDateStringToJapanFormat(dateOfRetirement)}
      />
      <Parameters name="退職理由" value={reason} />
      <Parameters
        name="残有給休暇日数"
        value={String(daysOfPaidLeaveRemaining)}
      />
    </div>
  )
}
