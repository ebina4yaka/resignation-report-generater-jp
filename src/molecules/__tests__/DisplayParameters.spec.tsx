import React from 'react'
import renderer from 'react-test-renderer'
import DisplayParameters from '../DisplayParameters'

jest.mock('../../atoms/Parameter', () => 'MockedParameters')

test('Snapshot test for DisplayParameters', () => {
  const tree = renderer.create(<DisplayParameters />).toJSON()
  expect(tree).toMatchSnapshot()
})
