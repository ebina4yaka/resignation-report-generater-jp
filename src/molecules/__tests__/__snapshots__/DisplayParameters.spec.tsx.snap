// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for DisplayParameters 1`] = `
<div
  className="makeStyles-root-1"
>
  <MockedParameters
    name="退職者"
    value=""
  />
  <MockedParameters
    name="会社名"
    value=""
  />
  <MockedParameters
    name="代表取締役"
    value=""
  />
  <MockedParameters
    name="退職日"
    value="Invalid Date"
  />
  <MockedParameters
    name="退職理由"
    value=""
  />
  <MockedParameters
    name="残有給休暇日数"
    value="0"
  />
</div>
`;
